### ISSUE TITLE
We should unwrap or expect on borsh `try_to_vec`

### ISSUE THREAD

**Issue by evgenykuzyakov** [nearprotocol/nearcore#1762]
Most of the code (all) that uses `try_to_vec` will never have an error from Borsh. Right now it's confusing which error a function can throw and some of them don't even need to return an `Error` at all.

If we replace `try_to_vec()?` with `try_to_vec().expect("borsh")` then the code going to be easier to handle. E.g.

https://github.com/nearprotocol/nearcore/blob/3ff069021a632807ec555d292503501c5e427ff5/chain/chain/src/validate.rs#L24

**Comment by evgenykuzyakov** [comment]
@mikhailOK

**Comment by bowenwang1996** [comment]
should we just implement something in borsh that unwraps the error?

**Comment by AnhTuDo1998** [comment]
Hi may I check if swapping propagation of error to expect statements still a good approach for this Issue?

**Comment by bowenwang1996** [comment]
> Hi may I check if swapping propagation of error to expect statements still a good approach for this Issue?

I think so. cc @matklad

**Comment by matklad** [comment]
@AnhTuDo1998 yes, we should generally add `.expect` to every try_to_vec. The devil is in details though -- we need to carefully check under which conditions this can actually panic (NaNs and collections longer than `u32::MAX` elements I think) and double-check that we can't hit such cases.
